pkg/util/fileoperator.go

```diff
@@ -38,10 +44,51 @@ func (r RealFileOperator) Write(filename string, data []byte, perm os.FileMode)
 	return err
 }
 
+// GetLineDetails return the line number and line content of matched string in file
+func (r RealFileOperator) GetLineDetails(filename, searchString string) (int, string, error) {
+	var line string
+	var i int
+	buffer, err := ioutil.ReadFile(filepath.Clean(filename))
+	if err != nil {
+		return -1, "", errors.Wrapf(err, "failed to read %s file", filename)
+	}
+	lines := strings.Split(string(buffer), "\n")
+	for i, line = range lines {
+		if strings.Contains(line, searchString) {
+			return i, line, nil
+		}
+	}
+	return -1, "", nil
+}
+
+// Updatefile updates the line number with the given string
+func (r RealFileOperator) Updatefile(filename, updatedVal string, index int, perm os.FileMode) error {
+	buffer, err := ioutil.ReadFile(filepath.Clean(filename))
+	if err != nil {
+		return errors.Wrapf(err, "failed to read %s file", filename)
+	}
+	lines := strings.Split(string(buffer), "\n")
+	lines[index] = updatedVal
```

Comment:
depending on `'index' variable, this fn is used.. instead, better to do string comparison for substring

Reply:
Updated